### PR TITLE
[cpp-qt-client] pass QString by const reference instead of by value

### PR DESCRIPTION
**Pull request by MartinDelille**
This PR change the signal signature.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@ravinikam @stkrwork @etherealjoy @muttleyxd

### CODE REVIEW

**Review by muttleyxd — APPROVED** [review]

**Comment by wing328** [comment]
Thanks for the PR.

> This PR change the signal signature.

so is it a breaking change?

**Comment by MartinDelille** [comment]
I would say it is not. Indeed, when trying to connect a signal to a slot with a different signature, it still works at least in the test I made.

I have two alternative in mind:
- rename the signal to break the compilation (replacing the EFull suffix by ErrorFull)
- add a new signal with another name and the new signature and deprecate the other.

What do you think about it ?

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-header.mustache [hunk 1]
```diff
@@ -93,7 +93,7 @@ signals:
 {{#operations}}{{#operation}}
     void {{nickname}}SignalE({{#returnType}}{{{.}}} summary, {{/returnType}}QNetworkReply::NetworkError error_type, QString error_str);{{/operation}}{{/operations}}
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-header.mustache:94]: I realize I should have change this signal signature too. Waiting for your thoughts about my last comment.

**Comment by wing328** [comment]
> add a new signal with another name and the new signature and deprecate the other.

I prefer this approach

**Comment by MartinDelille** [comment]
@muttleyxd @jimschubert @wing328 any feedback ?

**Comment by wing328** [comment]
restarted failed CI jobs for cpp-qt-client but no luck.

can you please take a look when you've time?

**Comment by MartinDelille** [comment]
@wing328 The error is quite strange since I am not using jurplel/install-qt-action v4...

I tried to add a name to disambiguate.

**Comment by wing328** [comment]
please resolve the merge conflicts when you've time.

**Comment by MartinDelille** [comment]
@wing328 Sorry I missed the notification. I fixed the Qt installation on Windows.

**Comment by MartinDelille** [comment]
@wing328 any chance to get this merged on the next release ?

**Comment by wing328** [comment]
thanks again for the PR, which has been merged.

sorry for the delay getting it merged.

**Pull request merged**